edward/inferences/variational_inference.py

```diff
@@ -87,38 +89,36 @@ def initialize(self, optimizer=None, var_list=None, use_prettytensor=False,
 
       self.summarize = tf.summary.merge_all(key=summary_key)
 
-    if optimizer is None:
+    learning_rate = 0.01
+    if optimizer is None or isinstance(global_step, tf.Variable):
       # Use ADAM with a decaying scale factor.
-      global_step = tf.Variable(0, trainable=False, name="global_step")
+      if not global_step:
```

Comment:
This logic could be flatter/less nested.

I'm also not sure if it works. For example, suppose `isinstance(global_step, tf.Variable)` is True and the user passes in `optimizer` as a `str`. Then the logic will create the decaying learning rate and exit the if-else clause without building any optimizer.